### PR TITLE
Add pagination to non-directory project views.

### PR DESCRIPTION
**Pull request by NullVoxPopuli**
https://developertown.visualstudio.com/SIL%20International%20Scriptoria/_workitems/edit/29205

 - adds pagination to all the project pages
 - makes the home page the tasks screen

### CODE REVIEW

File: source/SIL.AppBuilder.Portal.Frontend/src/ui/routes/project-directory/index.tsx [hunk 1]
```diff
@@ -49,8 +50,10 @@ export interface IOwnProps {
 
 export type IProps =
 & IOwnProps
+& ISortProps
 & IFilterProps
 & IDataProps
+& IPaginateProps
 & i18nProps;
```
Comment (giancorzo) [inline source/SIL.AppBuilder.Portal.Frontend/src/ui/routes/project-directory/index.tsx]: If using with translation also import i18nProps from same file

File: source/SIL.AppBuilder.Portal.Frontend/src/ui/routes/project-directory/index.tsx [hunk 2]
```diff
@@ -78,6 +81,7 @@ class ProjectDirectoryRoute extends React.Component<IProps> {
     } = this.props;
 
     const numProjects = projects && projects.length;
+    const isPaginationNeeded = numProjects > 19;
```
Comment (giancorzo) [inline source/SIL.AppBuilder.Portal.Frontend/src/ui/routes/project-directory/index.tsx]: Use constant

File: source/SIL.AppBuilder.Portal.Frontend/src/ui/routes/projects/archived-projects/index.tsx [hunk 3]
```diff
@@ -1,19 +1,65 @@
...
+export type IProps =
+& IPaginateProps
+& ISortProps
+& IProjectProps;
+
 
+class ArchivedProjects extends React.Component<IProps> {
   render() {
+    const { projects, toggleSort } = this.props;
+    const isPaginationNeeded = projects.length > 19;
```
Comment (giancorzo) [inline source/SIL.AppBuilder.Portal.Frontend/src/ui/routes/projects/archived-projects/index.tsx]: Same here use a constant

File: source/SIL.AppBuilder.Portal.Frontend/src/ui/routes/projects/header.tsx [hunk 4]
```diff
@@ -48,6 +52,6 @@ class Header extends React.Component<IOwnProps & i18nProps> {
 
 }
 
-export default compose(
+export default compose<IOwnProps, {}>(
   translate('translations'),
```
[the remark this answers is not in the document]
Comment (NullVoxPopuli) [reply source/SIL.AppBuilder.Portal.Frontend/src/ui/routes/projects/header.tsx]: not related to this PR. :)

File: source/SIL.AppBuilder.Portal.Frontend/src/ui/routes/index/index.tsx [hunk 5]
```diff
@@ -1,43 +1,18 @@
 import * as React from 'react';
```
[the remark this answers is not in the document]
Comment (NullVoxPopuli) [reply source/SIL.AppBuilder.Portal.Frontend/src/ui/routes/index/index.tsx]: while true, I don't know if we'll switch this at some point, for now this allows us slightly better display of intent.

[the remark this answers is not in the document]
Comment (NullVoxPopuli) [reply source/SIL.AppBuilder.Portal.Frontend/src/ui/routes/index/index.tsx]: imo

File: source/SIL.AppBuilder.Portal.Frontend/src/ui/routes/projects/header.tsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (giancorzo) [reply source/SIL.AppBuilder.Portal.Frontend/src/ui/routes/projects/header.tsx]: like you said the other day... If it's small you can do this little refactors

[the remark this answers is not in the document]
Comment (NullVoxPopuli) [reply source/SIL.AppBuilder.Portal.Frontend/src/ui/routes/projects/header.tsx]: that's true

**Pull request merged**